_posts/2016-07-14-official-docker-image.md

```diff
@@ -0,0 +1,170 @@
+
+### Volumes
+
+Portus' state is stored inside of a MariaDB database. This makes this Docker
+image stateless.
+
+### External database
+
+This image has a custom `init` script that takes care of configuring the external
+database.
+
+The script will keep trying to reach the database for 90 seconds. A 5 seconds
+pause is done after each failed attempt. The container will exit with an error
+message if the database is not reachable.
+
+The script will take care of the creation of the Portus database and its initial
+population via the usual Rails procedures.
+
+The database is automatically migrated whenever a new migration is introduced
+by the upstream project.
+
+### Secrets and certificates
+
+Portus requires both a SSL key and a certificate to serve its contents over
+HTTPS.
+These files must be located here:
+
+  * SSL certificate file: `/certificates/portus-ca.crt`
+  * SSL certificate key file: `/secrets/portus.key`
+
+The same key is also used to sign the JWT tokens issued to authenticate all the
+docker clients against the Registry.
+
+It's also required to add the certificate of the Registry when the latter one
+uses TLS to secure itself.
+The Registry certificate must be placed inside of `/certificates`, the `init`
+script of this image will automatically import it if it ends with the `.crt`
+extension.
+
+### Logging
+
+Both Portus' and Apache's messages are redirected to `stdout` and `stderr`. This
+makes it possible to handle the logs of this image in the usual "Docker ways".
+
+## Deployment
+
+It's possible to deploy this image using one of the existing orchestration
+solutions for Docker images.
+
+However we recommend the usage of [kubernetes](http://kubernetes.io/). We will
+soon release a reference implementation.
+
+### Environment variables
+
+The following environment variables must be defined to run the image:
+
+Security related settings:
+
+  * `PORTUS_SECRET_KEY_BASE`: you can generate it using `openssl rand -hex 64`
+  * `PORTUS_PORTUS_PASSWORD`: you can generate it using `openssl rand -hex 64`
+
+Database related settings:
+
+  * `MARIADB_SERVICE_HOST`: the host running the MariaDB database.
+  * `MARIADB_USER`: the database user to be used.
+  * `MARIADB_PASSWORD`: the password of the database user.
+  * `MARIADB_DATABASE`: the name of the Portus database.
+
+Deployment related settings:
+
+  * `PORTUS_MACHINE_FQDN_VALUE`: this is the fully qualified domain name of your
+    Portus instance (eg: `portus.example.com`).
+
+## Where to get it
+
+This image is available on the Docker Hub inside of the
+`https://hub.docker.com/r/opensuse/portus/` repository.
+
+This repository is an automated build, the original Dockerfile is
+defined inside of
+[this](https://github.com/openSUSE/docker-containers/tree/master/derived_images/portus)
+repository on GitHub.
+
+The image has different tags. The `head` tag points to the image using the Portus
+from the `master` branch.
+Each stable release of Portus has a dedicated tag.
```

Comment:
I would say that it should point to the latest stable release, but I don't have any strong feelings on this :smile: